src/main/java/org/folio/converter/QuickMarcToParsedRecordDtoConverter.java

```diff
@@ -163,11 +168,10 @@ private String restorePhysicalDescriptionsControlField(Map<String, Object> conte
   }
 
   private String restoreGeneralInformationControlField(Map<String, Object> contentMap) {
-    if (!contentMap.get(ELVL).toString().equals(Character.toString(leaderField.getImplDefined2()[0])) ||
-      !contentMap.get(DESC).toString().equals(Character.toString(leaderField.getImplDefined2()[1]))) {
+    if (!contentMap.get(ELVL).toString().equals(Character.toString(leaderString.charAt(ELVL_LEADER_POS))) ||
```

Comment:
Can contentMap be empty? Is it possible to refactor this logical expression to separate method with clearly readable name?

Reply:
Good idea, thank you!